sympy/tensor/indexed.py

```diff
@@ -359,7 +259,10 @@ def args(self):
             return self._args
 
     def _hashable_content(self):
-        return Expr._hashable_content(self) + (self._shape,)
+        if self._shape:
+            return super(IndexedBase, self)._hashable_content() + (self._shape,)
+        else:
+            return super(IndexedBase, self)._hashable_content()
```

Comment:
I think the bug is in the code down the line. `_hashable_content` should really be just fine with `None` values.
If it's a bug fix for an unrelated issue, then this should go into a separate PR. This makes it easier on future maintainers to find out why it was changed, and undo it once the actual bug is fixed. It would also avoid sidetracking the discussion about your tensor work with the discussion how and when to fix an obscure and currently non-reproducible bug.